docs/data_pipeline_procedure.md

```diff
@@ -0,0 +1,92 @@
+# Data Pipeline Procedure
+
+## Download Source Data
+   * Workspace Downloader
+     * Example usage
+        ```commandline
+        python src/loaders/workspace_downloader/workspace_downloader.py \
+          --env $env \
+          --workspace_id $workspace_id \
+          --kbase_collection $kbase_collection \
+          --source_ver $source_verion \
+          --token_filepath $token_filepath \
+          --retrieve_sample 
+        ```
+     * Generated files/metafiles
+       * downloaded files
+         ```text
+         sourcedata/WS/[env]/[workspace_id]/[UPA]/
+         ```
+       * Metadata files
+         ```text
+         sourcedata/WS/[env]/[workspace_id]/[UPA]/[UPA].meta
+         ```
+       * Softlinks for collections
+         ```text
+         collectionssource/[env]/[kbase_collection]/[source_ver]/[UPA]/
+         ```
+
+   * NCBI Downloader
+     * Example usage
+       ```commandline
+       python src/loaders/ncbi_downloader/gtdb.py \
+         --download_file_ext $download_file_ext \
+         --release_ver $release_ver
+       ```
+     * Generated files/metafiles
+       * downloaded files
+         ```text
+         sourcedata/NCBI/NONE/[ncbi_genome_id]/
+         ```
+       * Metadata files
+         ```text
+         None
+         ```
+       * Softlinks for collections
+         ```text
+         collectionssource/NONE/GTDB/[GTDB_release_ver]/[ncbi_genome_id]/
+         ```
+## Schedule Taskfarmer Jobs
+   * Example usage
+     ```commandline
+     python src/loaders/jobs/taskfarmer/task_generator.py \
+       --tool $tool \
+       --env $env \
+       --kbase_collection $kbase_collection \
+       --source_ver $source_ver \
+       --load_ver $load_ver \
+       --source_file_ext $source_file_ext \
+       --submit_job
+     ```
+   * Generated files/metafiles
+     * Taskfarmer job files
+       ```text
+       task_farmer_jobs/[job_dir]/
+       ```
+     * Taskfarmer jobs tracking file
+       ```text
+       task_farmer_jobs/task_info.jsonl
+       ```
+     * Tool result files
+       ```text
+       collectionsdata/[env]/[kbase_collection]/[load_ver]/[tool]/[batch_dir]/
+       ```
+       (`mash` outputs are saved in the source file directory)
+     * Tool result metadata file
+       ```text
+       collectionsdata/[env]/[kbase_collection]/[load_ver]/[tool]/[batch_dir]/genome_metadata.tsv
+       ```
+## Parse Tool Results
+   * Example usage
+     ```commandline
+     python src/loaders/genome_collection/parse_tool_results.py \
+       --env $env \
+       --kbase_collection $kbase_collection \
+       --source_ver $source_ver \
+       --load_ver $load_ver
```

Comment:
Strictly speaking if you know the load ver you know the source ver as well, or at least you should. Not sure how easy that is to figure out based on the files we have now, but in the DB it should be an explicit relation to my mind

Reply:
yea. I don't remember exactly why we keep 'source_ver' for parser. I think we can eliminate 'source_ver' from the parser script. It seems like it's only used to get the sample files which are also softlinked to collectionssource directory.

Reply:
Also to get mash files I think